deps/wazuh_testing/wazuh_testing/remote.py

```diff
@@ -27,81 +26,184 @@
+    """Creates a callback to detect syslog denied-ips.
+
+    Args:
+        syslog_ips (str): syslog denied-ips.
+
+    Returns:
+        callable: callback to detect this event.
+    """
     msg = fr"Message from \'{syslog_ips}\' not allowed. Cannot find the ID of the agent."
     return monitoring.make_callback(pattern=msg, prefix=monitoring.REMOTED_DETECTOR_PREFIX)
 
 
 def callback_invalid_value(option, value):
+    """Creates a callback to detect invalid values in ossec.conf file.
+
+    Args:
+        option (str): Wazuh manager configuration option.
+        value (str): Value of the configuration option.
+
+    Returns:
+        callable: callback to detect this event
+    """
     msg = fr"ERROR: \(\d+\): Invalid value for element '{option}': {value}."
     return monitoring.make_callback(pattern=msg, prefix=monitoring.REMOTED_DETECTOR_PREFIX)
 
 
 def callback_error_in_configuration(severity):
+    """Creates a callback to detect configuration error in ossec.conf file.
+
+    Args:
+        severity (str): ERROR or CRITICAL
+    Returns:
+        callable: callback to detect this event.
+    """
     msg = fr"{severity}: \(\d+\): Configuration error at '{WAZUH_CONF}'."
     return monitoring.make_callback(pattern=msg, prefix=monitoring.REMOTED_DETECTOR_PREFIX)
 
 
 def callback_error_invalid_port(port):
+    """Creates a callback to detect invalid port.
+
+    Args:
+        port (str): Wazuh manager port.
+    Returns:
+        callable: callback to detect this event.
+    """
     msg = fr"ERROR: \(\d+\): Invalid port number: '{port}'."
     return monitoring.make_callback(pattern=msg, prefix=monitoring.REMOTED_DETECTOR_PREFIX)
 
 
 def callback_ignored_invalid_protocol(protocol):
+    """Creates a callback to detect invalid protocol.
+
+    Args:
+        protocol (str): Wazuh manager protocol.
+    Returns:
+        callable: callback to detect this event.
+    """
     msg = fr"WARNING: \(\d+\): Ignored invalid value '{protocol}' for 'protocol'"
     return monitoring.make_callback(pattern=msg, prefix=monitoring.REMOTED_DETECTOR_PREFIX)
 
 
 def callback_error_getting_protocol():
+    """Creates a callback to detect if warning message is created when no valid protocol is provided.
+
+    Returns:
+        callable: callback to detect this event.
+    """
     msg = fr"WARNING: \(\d+\): Error getting protocol. Default value \(TCP\) will be used."
     return monitoring.make_callback(pattern=msg, prefix=monitoring.REMOTED_DETECTOR_PREFIX)
 
 
 def callback_warning_syslog_tcp_udp():
-    msg = fr"WARNING: \(\d+\): Only secure connection supports TCP and UDP at the same time. Default value \(TCP\) will be used."
+    """Creates a callback to detect if warning message is created when multiple protocol are provided using syslog.
+
+    Returns:
+        callable: callback to detect this event.
+    """
+    msg = fr"WARNING: \(\d+\): Only secure connection supports TCP and UDP at the same time. "
```

Comment:
It is necessary the last white space?

Reply:
Yes, it is log `wazuh-remoted` warning message style